crates/net/rpc/src/eth/api/mod.rs

```diff
@@ -43,17 +47,23 @@ where
     Client: BlockProvider + StateProviderFactory + 'static,
 {
     /// Creates a new, shareable instance.
-    pub fn new(client: Arc<Client>, pool: Pool) -> Self {
-        let inner = EthApiInner { client, pool };
+    pub fn new(client: Arc<Client>, pool: Pool, network: NetworkHandle) -> Self {
```

Comment:
please have a look at #979 

we only want traits here and not `NetworkHandle`

Reply:
Is is ok to add `chain_id()` to `NetworkInfo` or do you prefer to create another trait?

Reply:
yep let's add this